packages/@react-stately/layout/src/GridLayout.ts

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {DropTarget, DropTargetDelegate, ItemDropTarget, Key, Node} from '@react-types/shared';
+import {Layout, LayoutInfo, Rect, Size} from '@react-stately/virtualizer';
+
+export interface GridLayoutOptions {
+  /**
+   * The minimum item size.
+   * @default 200 x 200
+   */
+  minItemSize?: Size,
+  /**
+   * The maximum item size.
+   * @default Infinity
+   */
+  maxItemSize?: Size,
+  /**
+   * The minimum space required between items.
+   * @default 18 x 18
+   */
+  minSpace?: Size,
+  /**
+   * The maximum number of columns.
+   * @default Infinity
+   */
+  maxColumns?: number,
+  /**
+   * The thickness of the drop indicator.
+   * @default 2
+   */
+  dropIndicatorThickness?: number
+}
+
+export class GridLayout<T, O = any> extends Layout<Node<T>, O> implements DropTargetDelegate {
+  protected minItemSize: Size;
+  protected maxItemSize: Size;
+  protected minSpace: Size;
+  protected maxColumns: number;
+  protected dropIndicatorThickness: number;
+  protected itemSize: Size;
+  protected numColumns: number;
+  protected horizontalSpacing: number;
+  protected layoutInfos: LayoutInfo[];
+
+  constructor(options: GridLayoutOptions) {
+    super();
+    this.minItemSize = options.minItemSize || new Size(200, 200);
+    this.maxItemSize = options.maxItemSize || new Size(Infinity, Infinity);
+    this.minSpace = options.minSpace || new Size(18, 18);
+    this.maxColumns = options.maxColumns || Infinity;
+    this.dropIndicatorThickness = options.dropIndicatorThickness || 2;
+  }
+
+  validate(): void {
```

Comment:
This seems more like a "calculate" function.

Reply:
yeah this was existing in the Layout base class. But I agree, we should rename it before we release the major